### ISSUE TITLE
Saving cache can fail, breaking build (response 429/503)

### ISSUE THREAD

**Issue by vadi2** [actions/cache#467]
```
Post job cleanup.
/usr/bin/tar --posix --use-compress-program zstd -T0 -cf cache.tzst -P -C /Users/runner/work/Mudlet/Mudlet --files-from manifest.txt
Warning: Cache service responded with 429 during upload chunk.
/Users/runner/work/_actions/actions/cache/v2/dist/save/index.js:3305
                        throw new Error(`Cache upload failed because file read failed with ${error.message}`);
                        ^

Error: Cache upload failed because file read failed with EBADF: bad file descriptor, read
    at ReadStream.<anonymous> (/Users/runner/work/_actions/actions/cache/v2/dist/save/index.js:3305:31)
    at ReadStream.emit (events.js:210:5)
    at internal/fs/streams.js:167:12
    at FSReqCallback.wrapper [as oncomplete] (fs.js:470:5)
```

See https://github.com/Mudlet/Mudlet/runs/1435372417?check_suite_focus=true

Cache itself is rather simple,

```
    - name: (Linux/macOS) restore ccache
      uses: actions/cache@v2
      with:
        path: ${{runner.workspace}}/ccache
        key: ccache-${{matrix.os}}-${{matrix.compiler}}-${{matrix.qt}}-${{github.sha}}
        restore-keys: ccache-${{matrix.os}}-${{matrix.compiler}}-${{matrix.qt}}-
      if: runner.os != 'Windows'
```

**Comment by vadi2** [comment]
Sometimes a `503` can also be returned:

```
Post job cleanup.
/bin/tar --posix --use-compress-program zstd -T0 -cf cache.tzst -P -C /home/runner/work/Mudlet/Mudlet --files-from manifest.txt
Warning: uploadChunk (start: 637534208, end: 671088639) failed: Cache service responded with 503
/home/runner/work/_actions/actions/cache/v2/dist/save/index.js:3305
                        throw new Error(`Cache upload failed because file read failed with ${error.message}`);
                        ^

Error: Cache upload failed because file read failed with EBADF: bad file descriptor, read
    at ReadStream.<anonymous> (/home/runner/work/_actions/actions/cache/v2/dist/save/index.js:3305:31)
    at ReadStream.emit (events.js:210:5)
    at internal/fs/streams.js:167:12
    at FSReqCallback.wrapper [as oncomplete] (fs.js:470:5)
```

**Comment by pakatagoh** [comment]
I'm getting the same issue

```
Post job cleanup.
/bin/tar --posix -z -cf cache.tgz -P -C /home/pakata/actions-runner/_work/hackwagon-v2/hackwagon-v2 --files-from manifest.txt
Warning: uploadChunk (start: 0, end: 33554431) failed: Cache service responded with 503
/home/pakata/actions-runner/_work/_actions/actions/cache/v2/dist/save/index.js:3305
                        throw new Error(`Cache upload failed because file read failed with ${error.message}`);
                        ^

Error: Cache upload failed because file read failed with EBADF: bad file descriptor, read
    at ReadStream.<anonymous> (/home/pakata/actions-runner/_work/_actions/actions/cache/v2/dist/save/index.js:3305:31)
    at ReadStream.emit (events.js:210:5)
    at internal/fs/streams.js:167:12
    at FSReqCallback.wrapper [as oncomplete] (fs.js:470:5)
```

I'm running on a self-hosted server.

And this error only came about just a couple of days back

**Comment by ScottPierce** [comment]
Seeing the exact same thing

**Comment by akien-mga** [comment]
We're having the same issue at https://github.com/godotengine/godot, which receives a good 15 PRs per day and I often end up merging as many within a short timespan, which can easily trigger Too Many Requests (understandably).

Is there any known workaround? I don't mind if saving the cache fails, but I don't want my whole CI workflow to fail several times per day because of it, forcing me to restart build to get this nice green tick.

**Comment by dhadka** [comment]
@akien-mga Thanks!  Just making a note that this is related to https://github.com/actions/cache/issues/441.  I'll see if we can get 👀 on this issue.

(As a FYI, the `godot` repo is getting throttled due to the amount of content being uploaded.  There's a limit of 10 GBs uploaded every 5 minutes.  However, even if getting throttled, that shouldn't cause action / workflow to fail.)

**Comment by DemonWav** [comment]
This is frequently causing builds to fail for us as well https://github.com/minecraft-dev/MinecraftDev

**Comment by dhadka** [comment]
Merged #596 to fix this issue by catching and logging warnings.  The save step should no longer fail.  Fix in `v2.1.6`.
